src/test/cypress/support/pageobjects/OnlineEditorPage.ts

```diff
@@ -126,6 +131,39 @@ export class OnlineEditorPage {
     }
 }
 
+/**
+ * General method for entering, submitting and verifying something in the online editor.
+ */
+export function makeSubmissionAndVerifyResults(editorPage: OnlineEditorPage, packageName: string, submission: ProgrammingExerciseSubmission, verifyOutput: () => void) {
+    // We create an empty file so that the file browser does not create an extra subfolder when all files are deleted
+    editorPage.createFileInRootPackage('placeholderFile');
```

Comment:
We could remove it, but it doesn't affect the compilation in any way, so I don't see the value of this. In the best case nothing changes and in the worst case I add some flakiness into the test

Reply:
I agree, would it make more sense to not use the file approach but recreate the folder?
Because these e2e tests should test the way a user would do it, correct? And the user would probably delete the files which lead to the need to apparently recreate the folder.

While it doesn't change anything right now, it doesn't represent the way a user would do it and could therefore capture in the future only certain but not all parts.
Or do I overthink this here?

Reply:
My dream solution would be that no files is deleted at all and that the E2E test behaves like the user, but that is simply not possible in this scenario. A real user would open the programming exericse, open a file, and then insert his code into specific locations of the code template (like filling out the sort method etc). Cypress can't do this in a reliable way, so my previous solution was that when Cypress opens a file it deletes the content of the file by sending CTRL + A + DEL. That solution added a lot of flakiness. Sometimes cypress failed to execute that exact keypress combination in the right time and then the content was not deleted, which resulted in a build failure and as such a test failure.
That's why I moved to Cypress first deleting the file (to get rid of the content) and then creating a new file with the exact same name and writes the content in there. This is a lot more stable, but yeah as you mentioned it's not really what a user would do. Overall I still think that the test is very valuable in what it actually does test (file deletion, file creation, entering code via the ace code editor and checking the results)

Reply:
I see, thanks for the explanation. Good to know that there is thought behind this decision.
Though, I never heard of the combination of CTRL + A + DEL. What I know is CTRL + A to mark everything and then DEL to delete it. If you tried to execute the tripel, could it be the case that sometimes DEL was executed before A and therefore nothing was deleted?
Depending on a) whether you already investigated this or not and b) this only affects files in this PR or also other files, I would solve this in this PR, solve it in another PR or just ignore it then.

Reply:
Yeah I spent quite some time trying to get it stable. It sent CTRL + A and DEL with a delay (I also tried different delays). All options I tried with this approach (trying to clear the text via keyboard input) were flaky, so I went for the file deletion recreation approach.
I can have another look in the future, but I wouldn't change the approach now in this PR.

Reply:
Is there a reason why you didn't use the cypress clear function? https://docs.cypress.io/api/commands/clear
Sorry for the naive question. Just learning about cypress so... :D

Reply:
Yeah I tried that with cy.clear(). I also tried to use cy.invoke('text', ''). But both did not work. Cypress complains that the element is not clearable. To be honest I think we select the wrong element to type into. I found a [somewhat related thread](https://github.com/cypress-io/cypress/issues/1818). I will try the selector that the cypress developer proposed and see if that one is clearable.

Reply:
So I managed to clear the field with the selector proposed by the cypress developer, but it fails after the writing into the first file. The issue seems to be that multiple 'ace_text-input' elements exist after switching to another file. I will investigate this further in a followup PR since this also affects other specs (ProgrammingExerciseParticipation and ExamParticipation).

Reply:
Sound great. But amazing that we could make some progress don't that issue 👍